### PR TITLE
Added detailed error to 'az sf cluster create' when template validiation fails

### PR DESCRIPTION
**Pull request by lawrencegripper**
This provides a fix for #4790 which is a frustration when deploying Service Fabric clusters. It's not easy to find out what caused a validation error as detailed error messages aren't emitted. The PR adds these messages into the output to assist users. 

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

I'm unsure if this change warrants an entry, what do you think?

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

I haven't modified any commands. 

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by QingChenmsft** [comment]
@derekbekoe please unassign  from me, and assign to   @vaishnavk @juhacket

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py [hunk 1]
```diff
@@ -282,6 +284,17 @@ def new_cluster(client,
 
     return output_dict
 
+def _build_detailed_error(topError, outputList):
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py]: Variables should be underscore cased in Python

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py [hunk 2]
```diff
@@ -266,8 +266,10 @@ def new_cluster(client,
     validate_result = _deploy_arm_template_core(
         resource_group_name, template, parameters, deployment_name, 'incremental', True)
     if validate_result.error is not None:
-        raise CLIError("Template validates error \n'{}'".format(
-            validate_result.error))
+        errors_detailed = _build_detailed_error(validate_result.error, [])
+        error_message = "Error validating template. See below for more information. \n"
+        error_message += "".join(errors_detailed)
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py]: Would prefer not to have the `\n` manually entered for each line yourself then add the new lines as part of the join.
```
raise CLIError('\n'.join(error_messages))
```

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py [hunk 3]
```diff
@@ -282,6 +284,17 @@ def new_cluster(client,
 
     return output_dict
 
+def _build_detailed_error(topError, outputList):
+    if outputList:
+        outputList.append(" Inner Error - Code: \"{}\" Message: \"{}\" \n".format(topError.code, topError.message))
+    else:
+        outputList.append("Error - Code: \"{}\" Message: \"{}\" \n".format(topError.code, topError.message))
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py]: Can use single quotes for the string and then you don't need to escape the double quotes

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py (hunk 2 above)
Comment (lawrencegripper) [reply src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py]: Thanks, will do

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py (hunk 1 above)
Comment (lawrencegripper) [reply src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py]: Thanks, still fairly new to python so other language styles still creep in

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py (hunk 3 above)
Comment (lawrencegripper) [reply src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py]: Great, done

File: src/command_modules/azure-cli-servicefabric/setup.py [hunk 4]
```diff
@@ -16,7 +16,7 @@
     logger.warn("Wheel is not available, disabling bdist_wheel hook")
     cmdclass = {}
 
-VERSION = "0.0.6"
+VERSION = "0.0.7"
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-servicefabric/setup.py]: You can leave the version as 0.0.6 the current published version is 0.0.5 so 0.0.6 is okay.
https://pypi.python.org/pypi/azure-cli-servicefabric

File: src/command_modules/azure-cli-servicefabric/HISTORY.rst [hunk 5]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+0.0.7
++++++
+* Added details errors to validation response when creating cluster.
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-servicefabric/HISTORY.rst]: Move this to be under the 0.0.6 heading then remove the 0.0.7 heading

Comment (derekbekoe) [reply src/command_modules/azure-cli-servicefabric/HISTORY.rst]: Typo 'details' -> 'detailed'?

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py [hunk 6]
```diff
@@ -282,6 +283,17 @@ def new_cluster(client,
 
     return output_dict
 
+def _build_detailed_error(top_error, outputList):
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py]: `outputList` -> `output_list`

File: src/command_modules/azure-cli-servicefabric/HISTORY.rst (hunk 5 above)
Comment (lawrencegripper) [reply src/command_modules/azure-cli-servicefabric/HISTORY.rst]: Done, sry for the typo

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py (hunk 6 above)
Comment (lawrencegripper) [reply src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py]: Done

File: src/command_modules/azure-cli-servicefabric/setup.py (hunk 4 above)
Comment (lawrencegripper) [reply src/command_modules/azure-cli-servicefabric/setup.py]: Cool, rolled back

**Comment by lawrencegripper** [comment]
Thanks for the feedback, fixed those up now.

**Comment by lawrencegripper** [comment]
Thanks for the reviews @derekbekoe, any other changes needed on this one?

**Comment by derekbekoe** [comment]
@lawrencegripper We had a release a few days ago so please change version in setup.py to 0.0.7 and Modify your history change to a new heading under 0.0.7.
I closed and reopened the PR so it runs against the latest on the base branch.

**Comment by lawrencegripper** [comment]
Sorry for the delay @derekbekoe should all be done now. :)

**Comment by tjprescott** [comment]
Due to recent changes to the CLI core, you will need to rebase your changes against the latest in the dev branch.

**Comment by lawrencegripper** [comment]
Sure @tjprescott - all done.

**Comment by tjprescott** [comment]
Please check the CI. Your PR needs to bump the service fabric module version
```
There are changes to azure-cli-servicefabric and the current version 0.0.8 is already available on PyPI! Bump the version.
Changes are as follows:
* Updated SF module version number and history
* Updated SF module version and changelog. Refactored based on review.
* Added detailed error to 'az sf cluster create' when template validation fails #4790
Error(s) on azure-cli-servicefabric
The following modules have invalid versions:
azure-cli-servicefabric
```

**Comment by lawrencegripper** [comment]
@tjprescott Sorry, didn't see the notification of the build fail - all set.

**Pull request merged**